src/contrib/cluster/Akka.Cluster.Sharding/ClusterShardingGuardian.cs

```diff
@@ -303,8 +303,8 @@ internal static ReplicatorSettings GetReplicatorSettings(ClusterShardingSettings
                 .WithFallback(Context.System.Settings.Config.GetConfig("akka.cluster.distributed-data"));
             var configuredSettings = ReplicatorSettings.Create(config);
             var settingsWithRoles = configuredSettings.WithRole(shardingSettings.Role);
-            if (shardingSettings.RememberEntities)
-                return settingsWithRoles;
+            if (shardingSettings.RememberEntities && shardingSettings.RememberEntitiesStore == RememberEntitiesStore.DData)
+                return settingsWithRoles; // only enable durable keys when using DData for remember-entities
```

Comment:
Disable durable keys if we're not even using DData as the persistence mode for this remembered entity type.